webxrlayers-1.bs

```diff
@@ -908,6 +909,11 @@ section in the OpenGL ES 3.0 spec defines these limitations in more detail.
 
 Allocation of the resources for layers (such as memory) MUST be done through the same mechanism as WebGL.
 
+If an {{XRLayer}} is allocated with the {{RGBA}} or {{RGB}} {{XRLayerInit/colorFormat}}, its [=colorTextures=] MUST be exposed as {{RGBA}} or {{RGB}} to the {{WebGLRenderingContext}} context.
+However, the [=XR Compositor=] MUST treat the [=colorTextures=]'s pixels as if they were in the {{SRGB8_ALPHA8}} or {{SRGB8}} {{XRLayerInit/colorFormat}}.
+
+NOTE: this means that the [=XR Compositor=] MUST not do any gamma conversion from linear {{RGBA}} or {{RGB}} when it processes the [=colorTextures=]. Otherwise, the pixels in the final rendering will appear too bright which will not match the rendering on a regular 2D {{WebGLRenderingContext}} context.
```

Comment:
s/MUST not/MUST NOT/

There's a second problem also if the compositor erroneously treats the data as linear for compositing, but I'm not sure how to best fit that into the note. Maybe something like this?

NOTE: this means that the [=XR Compositor=] MUST NOT do any gamma conversion of the [=colorTextures=] pixel values *from* linear {{RGBA}} or {{RGB}} when it processes them for output on a device expecting nonlinear sRGB data. Conversely, if the [=XR Compositor=] requires linear data for blending or other internal processing, it MUST do a reverse gamma conversion of the [=colorTextures=] pixel values *to* linear. Otherwise, the pixels in the final rendering will appear too bright which will not match the rendering on a regular 2D {{WebGLRenderingContext}} context.

Reply:
I updated the upper casing of "MUST NOT" everywhere in the document.
I'm not aware of any compositor that works in linear light so I'm unsure if we should add it. We can discuss in the next meeting (or the face to face if you want to join).

Reply:
Technically, even simple interpolation or averaging from a GL_LINEAR texture lookup, for example when reading the projection layer during reprojection, will get the wrong result unless the calculation happens in linear space. As far as I understand, the GPU pretty much needs to know the correct color encoding for anything other than a 1:1 pixel copy. (Apparently the needed conversions are essentially free on modern GPUs?)

But this is a minor thing, it's implied from the new spec text. I can try to call in to the discussion if that works out, but I think we now have something solid already and I will try to refrain from obsessing over the remaining details....